src/lsp/cobol_parser/grammar.mly

```diff
@@ -2203,16 +2204,15 @@ let subscripts [@recovery []] [@symbol "<subscripts>"] [@cost 0] :=
  | "("; s = subscript_first; ")";                                 { [s] }
 
 (* Only for functions which name is also a keyword in COBOL *)
-let intrinsic_function_name :=
- | LENGTH;      { "LENGTH" }
- | RANDOM;      { "RANDOM" }
- | REVERSE;     { "REVERSE" }
- | SIGN;        { "SIGN" }
- | SUM;         { "SUM" }
+let intrinsic_function_name [@recover dummy_name] [@symbol "<intrinsic function name>"] :=
```

Comment:
Ideally we wouldn't need that conversion back to strings. Is there a way to avoid it?

Reply:
Same as for the typing of the token, we would need a special case in the type of `CallFunc {call_fun: <special_type_here>; ...}` if that's what you'd like

Reply:
Yes. But not `CallFunc`, rather `CallGenericIntrinsic of { intrinsic: generic_intrinsic_identifier; args: arguments }`.